app/models/stage_collection.rb

```diff
@@ -24,7 +24,7 @@ def previous_stage(stage)
   end
 
   def next_stage(stage)
-    find_stage(stage)&.first_valid_transition
+    find_stage(find_stage(stage)&.first_valid_transition)
```

Comment:
Ensure the result of this method is always a `Stage` object instead of a `symbol`.